packages/api/src/models/cleaningTaskModel.js

```diff
@@ -17,6 +17,67 @@ import Model from './baseFormatModel.js';
 import taskModel from './taskModel.js';
 
 class CleaningTaskModel extends Model {
+  // TODO: remove stub and update controller and FE to accept volume and weight
+  $parseDatabaseJson(json) {
+    // Remember to call the super class's implementation.
+    json = super.$parseDatabaseJson(json);
+    if (json.weight && json.weight_unit) {
+      json.product_quantity = json.weight;
+      json.product_quantity_unit = json.weight_unit;
+    } else if (json.volume && json.volume_unit) {
+      json.product_quantity = json.volume;
+      json.product_quantity_unit = json.volume_unit;
+    } else if (!json.volume && !json.weight) {
+      json.product_quantity = null;
+      json.product_quantity_unit = null;
+    }
+    // Database checks prevent quantity && !quantity_unit
+    delete json.weight;
+    delete json.weight_unit;
+    delete json.volume;
+    delete json.volume_unit;
+    return json;
+  }
+
+  // TODO: remove stub and update controller and FE to accept volume and weight
+  $formatDatabaseJson(json) {
+    // Remember to call the super class's implementation.
+    json = super.$formatDatabaseJson(json);
+    const weightUnits = ['g', 'lb', 'kg', 't', 'mt', 'oz'];
+    const volumeUnits = ['l', 'gal', 'ml', 'fl-oz'];
+    const defaultAction = (json) => {
+      json.volume = json.product_quantity;
+      //Database previously defaulted to 'l'
+      json.volume_unit = 'l';
+      json.weight = null;
+      json.weight_unit = null;
+      delete json.product_quantity;
+      delete json.product_quantity_unit;
```

Comment:
I think these can be done before returning json(line 83) too!

Reply:
true.. haha